### PR TITLE
DSDEGP-2151 & 2238: Export progress bar fix & Overall download progress display doesn't remove items that are transferred

### PR DESCRIPTION
**Pull request by LeoForconesi**
### Description

_All filters used to count and display progress have been modified in order to reflect real progress._
_Also some modifications are done here for Overall download progress to display items that are transferred._

_**Jira Stories included in this Code:**_

- [DSDEGP-2151](https://broadinstitute.atlassian.net/browse/DSDEGP-2151) : Overall download progress display doesn't remove items that are transferred

- [DSDEGP-2238](https://broadinstitute.atlassian.net/browse/DSDEGP-2238) : Export Progress bar not working as expected

----

### Checklist

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [] **Submitter**: Ensure that you have added or updated tests
- [x] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=189431877).
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: src/app/file-manager/reducers/transferables.reducer.ts [hunk 1]
```diff
@@ -8,111 +8,114 @@ import { Type } from '@app/file-manager/models/type';
 export type Action = TransferablesActions.All;
 
 export class TransferableState {
-  items: Item[];
+    items: Item[];
```
Comment (pshapiro4broad) [inline src/app/file-manager/reducers/transferables.reducer.ts]: Looks like a re-indent issue; I believe we're using 2 space indent for these files, not 4 space.

Comment (LeoForconesi) [reply src/app/file-manager/reducers/transferables.reducer.ts]: indeed, we are using 2 space indent, this is now fixed.

File: src/app/file-manager/file-download-modal/file-download-modal.component.ts [hunk 2]
```diff
@@ -12,6 +12,12 @@ import { Router } from '@angular/router';
 import { ItemStatus } from '@app/file-manager/models/item-status';
 import { TransferablesGridComponent } from '@app/file-manager/transferables-grid/transferables-grid.component';
 import { PreflightService } from '../services/preflight.service';
+import { FilterSizePipe } from '../filters/filesize-filter';
```
Comment (pshapiro4broad) [inline src/app/file-manager/file-download-modal/file-download-modal.component.ts]: remove unused imports, `FilterSizePipe`, `Observable`, `BehaviorSubject`

File: src/app/file-manager/file-download-modal/file-download-modal.component.ts [hunk 3]
```diff
@@ -135,4 +143,13 @@ export class FileDownloadModalComponent implements OnInit {
   selectedFiles() {
     return this.preflightService.selectedFiles;
   }
+
+  updateCurrentBatch() {
+    const items = new FilesDatabase(this.store).data.filter(item => item.type === Type.DOWNLOAD && item.currentBatch);
```
Comment (pshapiro4broad) [inline src/app/file-manager/file-download-modal/file-download-modal.component.ts]: Is there a way to avoid the code duplication here, such as by making this method accept `type`?

File: src/app/file-manager/models/item.ts [hunk 4]
```diff
@@ -26,10 +26,11 @@ export class Item {
   indeterminate: boolean;
   namespace: string;
   children?: Item[];
+  currentBatch: boolean;
 
   constructor(id, name, created, updated, size, mediaLink, path, destination,
     type, status, bucketName, prefix, delimiter, preserveStructure, open,
-    workspaceName, displayName, namespace) {
+    workspaceName, displayName, namespace, currentBatch) {
```
Comment (pshapiro4broad) [inline src/app/file-manager/models/item.ts]: These arguments should all have declared types, e.g. `currentBatch: boolean`

File: src/app/file-manager/services/gcs-api.service.ts [hunk 5]
```diff
@@ -160,7 +160,7 @@ export class GcsApiService extends GcsService {
         'Authorization': 'Bearer ' + SecurityService.getAccessToken()
       })
     };
-    return this.http.post(url, null, httpOptions);
+    return this.http.post(url, null, httpOptions).retry(1);
```
Comment (pshapiro4broad) [inline src/app/file-manager/services/gcs-api.service.ts]: Why is this change necessary? A comment would be helpful.

File: src/app/file-manager/transferables-grid/transferables-grid.component.ts [hunk 6]
```diff
@@ -28,10 +28,12 @@ import { WarningModalComponent } from '../warning-modal/warning-modal.component'
 
 export class TransferablesGridComponent implements OnInit, AfterViewInit {
   static isExporting: Boolean = false;
+  static firstIteration: Boolean = true;
   @ViewChild(MatPaginator) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
-  displayedColumns = ['name', 'size', 'status', 'progress', 'actions'];
+  @ViewChild('pbg') pbg: ProgressBar;
```
Comment (pshapiro4broad) [inline src/app/file-manager/transferables-grid/transferables-grid.component.ts]: This appears to be unused; if so, it (and the copy in HTML) should be removed.

File: src/app/file-manager/transferables-grid/transferables-grid.component.ts [hunk 7]
```diff
@@ -28,10 +28,12 @@ import { WarningModalComponent } from '../warning-modal/warning-modal.component'
 
 export class TransferablesGridComponent implements OnInit, AfterViewInit {
   static isExporting: Boolean = false;
+  static firstIteration: Boolean = true;
```
Comment (pshapiro4broad) [inline src/app/file-manager/transferables-grid/transferables-grid.component.ts]: Why does this need to be `static`? It looks like it and `isExporting` can be converted to fields.

**Review by pshapiro4broad — COMMENTED** [review]
A few minor comments. When I tried this out, I did notice this behavior:

- start a long download, cancel it 
- at this point the Downloaded progress bar is partly full, depending on how much data has been downloaded
- switch to the Download view, then switch back to Status
- the progress bar is now empty

This isn't a major issue, but I was expecting that the progress bar and the "x out of y files" message would be updated when a cancel occurs, since this status no longer applies to the current download. I was expecting that bar would be empty, and the message would say "0 out of 0 files".

File: src/app/file-manager/services/gcs-api.service.ts (hunk 5 above)
Comment (LeoForconesi) [reply src/app/file-manager/services/gcs-api.service.ts]: I added this change in order to minimize impact from the Out of resources issue that we are experiencing from Chromium limitations when exporting a large amount of files between Google Buckets. 

By default angular's http client will retry multiple times before dropping this request. Meanwhile, the app will continue sending more posts from other exports (these might fail too), causing a request memory overflow, leading to chromium's error.

The idea here is to retry only once if the request fails, stopping any request flood.

File: src/app/file-manager/transferables-grid/transferables-grid.component.ts (hunk 7 above)
Comment (LeoForconesi) [reply src/app/file-manager/transferables-grid/transferables-grid.component.ts]: The reason these are not local variables, is because their values are not persisting through the app execution.

When receiving info from observables we get a new instance from the component, which redefines all these variables, needed to reflect changes in the Ui. 

We were discussing a possible solution that is implement Redux store in order to reflect the app state allowing us to remove every static variable, but at the moment this would be a temporal solution.

File: src/app/file-manager/file-download-modal/file-download-modal.component.ts (hunk 2 above)
Comment (LeoForconesi) [reply src/app/file-manager/file-download-modal/file-download-modal.component.ts]: Unused imports were removed.

File: src/app/file-manager/transferables-grid/transferables-grid.component.ts (hunk 6 above)
Comment (LeoForconesi) [reply src/app/file-manager/transferables-grid/transferables-grid.component.ts]: Sorry about that, this variable is removed.

**Comment by LeoForconesi** [comment]
@pshapiro4broad The behavior you saw with downloads is registered in Jira, and will be solved soon. Thanks for the heads up!

File: src/app/file-manager/transferables-grid/transferables-grid.component.ts (hunk 7 above)
Comment (pshapiro4broad) [reply src/app/file-manager/transferables-grid/transferables-grid.component.ts]: It's OK to leave these as statics for now. Maybe add a comment about how a new component is created each time but this state has a longer lifetime than the component. The goal would be to put the properties on an object that has the correct lifetime, maybe the Store is the best place.

File: src/app/file-manager/services/gcs-api.service.ts (hunk 5 above)
Comment (pshapiro4broad) [reply src/app/file-manager/services/gcs-api.service.ts]: That's good, can you add a comment that describes this? Reading the code I had assumed that, by default, no retries would occur and the change was to request a retry.

**Review by pshapiro4broad — APPROVED** [review]
